### ISSUE TITLE
RecyclerView sometimes doesn't refresh when adding a new item

### ISSUE THREAD

**Issue by ZeGerm4n** [udacity/kotlin-notepad#4]
The sample project works fine on an emulator running Android Oreo, but has an annoying bug on real devices like Samsung Galaxy S3(LineageOS 14) and Sony Xperia S(nAOSP 7.1). When running the sample project on these older devices, the list does not always update when adding a new item. It is really annoying to stumble over a bug in a udacity sample project, especially when you are a newbie to programming like myself.

**Comment by dnbit** [comment]
@ZeGerm4n 
You are using custom ROMs in your devices. In order to verify that everything would work properly in every single non-stock ROM and device combination, we would need to have every physical device and non-stock ROM combination, plus various lifetimes of testing.

Because the Android platform is open and implementers are free to make a lot of decisions on their own, sometimes there are differences between ROMs that can be difficult to debug.

**Comment by ZeGerm4n** [comment]
Just tried running the project on Emulator images with API versions 22, 23, 25, 27. All of them where default set-ups of virtual google Nexus devices available for selection in AVD Manager. I noticed that all of these had this issue. The Oreo one (of which I previously thought it had no issues) seems to encounter this problem too, but less frequently. Seems like it has nothing to do with the custom ROMs I use :/ I would be very grateful if someone could check what might cause the problem, as of i am not familiar enough with android development to do it myself.

**Comment by ZeGerm4n** [comment]
In case someone encounters the same problem:
On the edge of despair I did following... I "fixed" the behaviour by using startActivityForResult() to call CreateActivity. In CreateActivity I query the new item count in the database after inserting a new item and pass that integer in setResult() before finishing the Activity. Back in MainActivity, in onActivityResult() I compare the the item count of the database with the item count in RecyclerViewAdapter and finally call refresh if those values are unequal

**Comment by dnbit** [comment]
That's an overkill @ZeGerm4n You should not need to do that.
I have tried in various versions and it works for me. It seems that I cannot reproduce it.
In which branch do you see the error? Any tips to reproduce it?

**Comment by ZeGerm4n** [comment]
I encountered the problem on the 'final' branch. Last time I could reproduce it was by killing the App completely then restarting it and trying to add multiple items. After creating 1-3 items the newest one would only be displayed after adding another one. I just tried to reproduce it again, but the closest I could get to above described behaviour was that the RecyclerView took a second to display the newly created item. I'm baffled, have no idea what could have been the culprit and why I cannot reproduce that error anymore o_O anyway thank you for looking into it
